crates/iota-sdk-ffi/src/types/signature.rs

```diff
@@ -1,5 +1,268 @@
+    }
+
+    pub fn as_multisig_opt(&self) -> Option<Arc<MultisigAggregatedSignature>> {
+        if let iota_types::UserSignature::Multisig(sig) = self.0.clone() {
+            Some(Arc::new(MultisigAggregatedSignature(sig)))
+        } else {
+            None
+        }
+    }
+
+    pub fn as_multisig(&self) -> Arc<MultisigAggregatedSignature> {
+        self.as_multisig_opt().expect("not a multi-signature")
+    }
+
+    pub fn is_zklogin(&self) -> bool {
+        matches!(self.0, iota_types::UserSignature::ZkLogin(_))
+    }
+
+    pub fn as_zklogin_opt(&self) -> Option<Arc<ZkLoginAuthenticator>> {
+        if let iota_types::UserSignature::ZkLogin(sig) = self.0.clone() {
+            Some(Arc::new(ZkLoginAuthenticator(*sig)))
+        } else {
+            None
+        }
+    }
+
+    pub fn as_zklogin(&self) -> Arc<ZkLoginAuthenticator> {
+        self.as_zklogin_opt().expect("not a zklogin authenticator")
+    }
+
+    pub fn is_passkey(&self) -> bool {
+        matches!(self.0, iota_types::UserSignature::Passkey(_))
+    }
+
+    pub fn as_passkey_opt(&self) -> Option<Arc<PasskeyAuthenticator>> {
+        if let iota_types::UserSignature::Passkey(sig) = self.0.clone() {
+            Some(Arc::new(sig.into()))
+        } else {
+            None
+        }
+    }
+
+    pub fn as_passkey(&self) -> Arc<PasskeyAuthenticator> {
+        self.as_passkey_opt().expect("not a passkey signature")
+    }
+}
+
+/// A basic signature
+///
+/// This enumeration defines the set of simple or basic signature schemes
+/// supported by IOTA. Most signature schemes supported by IOTA end up
+/// comprising of a at least one simple signature scheme.
+///
+/// # BCS
+///
+/// The BCS serialized form for this type is defined by the following ABNF:
+///
+/// ```text
+/// simple-signature-bcs = bytes ; where the contents of the bytes are defined by <simple-signature>
+/// simple-signature = (ed25519-flag ed25519-signature ed25519-public-key) /
+///                    (secp256k1-flag secp256k1-signature secp256k1-public-key) /
+///                    (secp256r1-flag secp256r1-signature secp256r1-public-key)
+/// ```
+///
+/// Note: Due to historical reasons, signatures are serialized slightly
+/// different from the majority of the types in IOTA. In particular if a
+/// signature is ever embedded in another structure it generally is serialized
+/// as `bytes` meaning it has a length prefix that defines the length of
+/// the completely serialized signature.
+#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
+pub struct SimpleSignature(pub iota_types::SimpleSignature);
+
+#[uniffi::export]
+impl SimpleSignature {
+    pub fn scheme(&self) -> SignatureScheme {
+        self.0.scheme()
+    }
+
+    fn to_bytes(&self) -> Vec<u8> {
+        self.0.to_bytes()
+    }
+
+    pub fn is_ed25519(&self) -> bool {
+        matches!(self.0, iota_types::SimpleSignature::Ed25519 { .. })
+    }
+
+    pub fn ed25519_sig_opt(&self) -> Option<Arc<Ed25519Signature>> {
```

Comment:
Name not consistent with the rest, missing the `as_` prefix. Same for following methods.